qiskit/circuit/instructionset.py

```diff
@@ -95,7 +95,17 @@ class InstructionSet:
 
     __slots__ = ("_instructions", "_requester")
 
-    def __init__(
+    @deprecate_arg(
+        "circuit_cregs",
+        since="0.19.0",
+        additional_msg=(
+            "Instead, pass a complete resource requester with the 'resource_requester' argument. "
+            "(The classical registers are insufficient to access all classical resources in a "
+            "circuit, as there may be loose classical bits as well. It can also cause integer "
+            "indices to be resolved incorrectly if any registers overlap.)"
```

Comment:
super nitpicky, but the parenthetical sentences here end up being longer than the actual deprecation message, which feels pretty weird to me.  It's either important information for the user, in which case it needn't be parenthesised, or it's not and it can be a code/docs-only comment instead.  I don't particularly mind which.